src/httpserver.cpp

```diff
@@ -374,7 +377,11 @@ static bool HTTPBindAddresses(struct evhttp* http)
         for (const std::string& strRPCBind : gArgs.GetArgs("-rpcbind")) {
             uint16_t port{http_port};
             std::string host;
-            SplitHostPort(strRPCBind, port, host);
+            if (!SplitHostPort(strRPCBind, port, host)) {
```

Comment:
Code review ACK 51ba0e61273bec02833329ee6ac8fa219679cefa

(Not important, but another idea to make this commit smaller and clearer: I think it would be good to move the change in `src/httpserver.cpp` to a separate followup commit, because while it's a good change that improves code quality and error reporting, it is not really a part of the bugfix, and is more tangentially related to it.)

Reply:
Thanks.  Yes, that's correct (movement of CheckHostPortOptions() handles the bug before HTTPBindAddresses()). It's cleaner to break this out into its own commit. Updated.